### DISCUSSION TITLE
How can we change OrbitControls to PointerLockControls in model-viewer?

### DISCUSSION THREAD

**Discussion by ikarus-pritish** [google/model-viewer#4225] [Q&A] [answered]
I have a requirement where I am adding scenes to the web page. The scene is a room and PointerLockControls would perform better compared to OrbitControls to experience a completely natural interaction with the scene. Is there any way we can add that to model-viewer?

PS: PointerLockControls - https://threejs.org/docs/#examples/en/controls/PointerLockControls
https://threejs.org/examples/#misc_controls_pointerlock

**Comment by elalish** [comment, marked as answer]
You can not add the `camera-controls` attribute and then implement your own pointer listeners and update the `cameraOrbit` and such as you see fit.

**Reply by ikarus-pritish** [reply]
Hi @elalish. Thank you for replying. If I understand it correctly, you are saying that I can omit `camera-controls` attribute and add pointer listeners. I will try this and update here.
